**Laboratorio 6**

María Paula Angulo Collantes

Danna Gabriela Castro

In [16]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
df = pd.read_csv('bicicletas_compartidas_bogota.csv')

**Punto 1**

In [17]:
sub = df.dropna(subset=['distancia_km','duracion_min'])
x = sub['distancia_km'].to_numpy()
y = sub['duracion_min'].to_numpy()

cov = np.cov(x, y, ddof=1)[0, 1]

r = cov / (x.std(ddof=1) * y.std(ddof=1))
print(f"Pearson r: {r:.4f}")


Pearson r: 0.7217


**1.b** 

**Dirección:** positiva, a mayor distancia recorrida, tiende a haber mayor duración del viaje.

**Intensidad:** moderada-alta. Un r de 0.72 no es una correlación perfecta, pero sí es considerablemente fuerte para datos de comportamiento humano/urbano, donde hay mucha variabilidad.

**Significado práctico:** R²≈0.52. El 52% de la variación en duración se explica linealmente por la distancia recorrida. El 48% restante depende de otros factores no capturados aquí.

**1.c**

No es correcto afirmarlo únicamente con base en la correlación, porque esta muestra asociación, no causalidad. Aunque existe una relación física razonable entre distancia y duración, podrían influir otras variables como el clima o el tipo de bicicleta. Por eso, lo adecuado es decir que existe una asociación positiva entre ambas variables, no que la distancia causa directamente la duración.

**1.d**

**Punto 2**

**Punto 3**

**3.a**

**3.b**


**3.c**

**Punto 4**

**Punto 5**

¿Cambian las diferencias entre condiciones climáticas?

¿Por qué comparar todos los viajes directamente podría resultar engañoso?

¿Qué papel puede estar desempeñando distancia km?

¿Podemos concluir que el clima causa cambios en la duración del viaje?

**Punto 6**